### PR TITLE
B 21566 - Create UBP, ISLH, IOSFSC, IDSFSC Services

### PR DESCRIPTION
**Pull request by brianmanley-caci**
##[INTEGRATION PR]
(https://github.com/transcom/mymove/pull/14072)
(https://github.com/transcom/mymove/pull/14161)

## [Agility ticket]
(https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1023571&RoomContext=TeamRoom%3A1007470&concept=TeamRoom)

## Summary
This story is to add the new UBP service item which is replacing these now obsolete service items: ICOUB, IOOUB, IOCUB, NSTUB. We are also adding the new ISLH service item which is replacing: IOCLH, ICOLH, IOOLH, NSTH. The obsolete services are removed from service_params, and re_services. All rows for pricing of the obsolete services in re_intl_prices are now pointed to the new replacment service (UBP or ISLH). The re_service_items table is added to associate a service to a market/shipment type combo.

Additionally we are adding IOSFSC and IDSFSC to re_services and re_service_items.

Is there anything you would like reviewers to give additional scrutiny? Maximum scrutiny on all changes.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

###Database

1. Run the migrations.
2. Verify that IOCUB, ICOUB, IOOUB, NSTUB, IOCLH, ICOLH, IOOLH, NSTH are no longer in the re_services table.
3. Verify that UBP, ISLH, IOSFSC, and IDSFSC are in the re_services_table.
4. Verify that there are rows tied to UBP and ISLH in the re_intl_prices table.
5. Verify that there are rows tied to UBP, ISLH, IOSFSC, and IDSFSC in the re_service_items_table.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: migrations/app/schema/20241023130404_create_re_service_items.up.sql [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+(id, code, name, created_at, updated_at, priority, service_location)
+VALUES('9f3d551a-0725-430e-897e-80ee9add3ae9', 'ISLH', 'International Shipping & Linehaul', now(), now(), 99, 'B');
+
+INSERT INTO public.re_services
+(id, code, name, created_at, updated_at, priority, service_location)
+VALUES('f75758d8-2fcd-40ba-9432-3ff3032a71d1', 'POEFSC', 'International POE Fuel Surcharge', now(), now(), 99, 'O');
+
+INSERT INTO public.re_services
+(id, code, name, created_at, updated_at, priority, service_location)
+VALUES('388115e8-abe9-441d-96cf-a39f24baa0a3', 'PODFSC', 'International POD Fuel Surcharge', now(), now(), 99, 'D');
```
[the remark this answers is not in the document]
Comment (brianmanley-caci) [reply migrations/app/schema/20241023130404_create_re_service_items.up.sql:49]: They were included with the setup of the re_service_items data. I have added them to the story ac.

File: pkg/services/mto_service_item/mto_service_item_creator_test.go [hunk 2]
```diff
@@ -1511,11 +1511,8 @@ func (suite *MTOServiceItemServiceSuite) TestCreateDestSITServiceItem() {
 			Status:           models.MTOServiceItemStatusSubmitted,
 		}
 
-		createdServiceItems, _, err := creator.CreateMTOServiceItem(suite.AppContextForTest(), &serviceItemDDFSIT)
-		suite.Nil(createdServiceItems)
-		suite.Error(err)
-		suite.IsType(apperror.NotFoundError{}, err)
-		suite.Contains(err.Error(), "service code")
+		_, _, err := creator.CreateMTOServiceItem(suite.AppContextForTest(), &serviceItemDDFSIT)
+		suite.NoError(err)
```
Comment (ryan-mchugh) [inline pkg/services/mto_service_item/mto_service_item_creator_test.go:1515]: Appears that this actually passes now, so may want to update the test description from "Failure"

**Review by traskowskycaci — APPROVED** [review]
One diff in playwright/tests/office/servicescounseling/servicesCounselingFlows.spec.js from https://github.com/transcom/mymove/pull/14161/files but change alreadys exists in main, so g2g and otherwise matches int PRs. Nice!

File: pkg/testingsuite/pop_suite_test.go [hunk 3]
```diff
@@ -164,28 +139,22 @@ func (suite *PreloadedPopSuite) TestRunAlt() {
 
 		var foundReService models.ReService
 		for _, reservice := range suite.ReServices {
-			err := suite.DB().Find(&foundReService, reservice.ID)
+			err := suite.DB().Where("code = $1", reservice.Code).First(&foundReService)
 			suite.NoError(err, "Reservice %s not found", reservice.Code)
 		}
 		// Add a DUCRT ReService, this should not exist outside this subtest
-		factory.BuildReServiceByCode(suite.DB(), models.ReServiceCodeDUCRT)
+		factory.FetchReServiceByCode(suite.DB(), models.ReServiceCodeDUCRT)
```
Comment (ryan-mchugh) [inline pkg/testingsuite/pop_suite_test.go:146]: Does DUCRT now exist outside the subtest? (I'm gathering that codes should not be created with the ReService, just retrieved. correct?)

File: pkg/testingsuite/pop_suite_test.go [hunk 4]
```diff
@@ -164,28 +139,22 @@ func (suite *PreloadedPopSuite) TestRunAlt() {
...
-		//                   The one new reService added in the subtest above is NOT found
 		var foundReService models.ReService
 		for _, reservice := range suite.ReServices {
-			err := suite.DB().Find(&foundReService, reservice.ID)
+			err := suite.DB().Where("code = $1", reservice.Code).First(&foundReService)
 			suite.NoError(err, "Reservice %s not found", reservice.Code)
 		}
-
-		var foundReServices []models.ReService
-		err := suite.DB().Where("code = ?", models.ReServiceCodeDUCRT).All(&foundReServices)
```
Comment (ryan-mchugh) [inline pkg/testingsuite/pop_suite_test.go:186]: I.E. why was this removed?

File: pkg/testingsuite/pop_suite_test.go (hunk 3 above)
Comment (brianmanley-caci) [reply pkg/testingsuite/pop_suite_test.go:146]: Yes. All ReServices are readonly from the app perspective now. DUCRT will already exist in the DB based off the migration scripts.

Comment (ryan-mchugh) [reply pkg/testingsuite/pop_suite_test.go:146]: Ok makes sense. Comments & tests could be updated to reflect -- but not holding this PR back for it. Something to think about in downstream potentially.

**Review by ryan-mchugh — APPROVED** [review]

File: pkg/testingsuite/pop_suite_test.go (hunk 4 above)
Comment (brianmanley-caci) [reply pkg/testingsuite/pop_suite_test.go:186]: > I.E. why was this removed?

It appears this check here was to verify that DUCRT wasn't found. It will always exist in the DB now since re_services isn't truncated, I think I felt it would be fine to remove this part of the test.

**Review by msaki-caci — APPROVED** [review]

**Comment by traskowskycaci** [comment]
> Requesting that the changes from [this commit](https://github.com/transcom/mymove/pull/14166/commits/11dd99075108bb858cb226d3a01c961c9bbfb99c) get brought into integration before merging this PR into main. Feel free to dismiss this if they're already there and I missed it

Confirmed they were brought into [this INT PR](https://github.com/transcom/mymove/pull/14171), looks good!

**Review by traskowskycaci — APPROVED** [review]

File: pkg/services/mto_service_item/mto_service_item_creator_test.go (hunk 2 above)
Comment (brianmanley-caci) [reply pkg/services/mto_service_item/mto_service_item_creator_test.go:1515]: Fixed and applied to integration.

**Review by WeatherfordAaron — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]
